server/src/test/java/org/apache/cassandra/sidecar/db/SystemAuthDatabaseAccessorTest.java

```diff
@@ -48,4 +61,36 @@ void testAccessorThrowsWhenTableNotFound()
         .isInstanceOf(SchemaUnavailableException.class)
         .hasMessage("Table system_auth.identity_to_role does not exist");
     }
+
+    @Test
+    void testReadingMemberOf()
+    {
+        SystemAuthSchema mockSystemAuthSchema = mock(SystemAuthSchema.class);
+        CQLSessionProvider mockCqlSessionProvider = mock(CQLSessionProvider.class);
+        Session mockSession = mock(Session.class);
+        PreparedStatement mockPreparedStatement = mock(PreparedStatement.class);
+        when(mockSystemAuthSchema.allRoles()).thenReturn(mockPreparedStatement);
+        when(mockCqlSessionProvider.get()).thenReturn(mockSession);
+        ResultSet mockResult = mock(ResultSet.class);
+        List<Row> rows = new ArrayList<>();
+        Row mockRow1 = mock(Row.class);
+        when(mockRow1.getString("role")).thenReturn("super_role");
+        when(mockRow1.getBool("is_superuser")).thenReturn(true);
+        when(mockRow1.getSet("member_of", String.class)).thenReturn(Set.of());
+        Row mockRow2 = mock(Row.class);
+        when(mockRow2.getString("role")).thenReturn("non_super_role");
+        when(mockRow2.getBool("is_superuser")).thenReturn(false);
+        when(mockRow2.getSet("member_of", String.class)).thenReturn(Set.of("super_role"));
+        rows.add(mockRow1);
+        rows.add(mockRow2);
+        when(mockResult.all()).thenReturn(rows);
+        when(mockSession.execute((Statement) any())).thenReturn(mockResult);
+        SystemAuthDatabaseAccessor systemAuthDatabaseAccessor = new SystemAuthDatabaseAccessor(mockSystemAuthSchema,
+                                                                                               mockCqlSessionProvider,
+                                                                                               new PermissionFactoryImpl());
+        Map<String, Boolean> superUserCache = systemAuthDatabaseAccessor.findAllRolesToSuperuserStatus();
+        assertThat(superUserCache.size()).isEqualTo(2);
+        assertThat(superUserCache.get("super_role")).isTrue();
+        assertThat(superUserCache.get("non_super_role")).isTrue();
+    }
```

Comment:
I am not sure this test is meaningful.. It just mocks everything and mock is based on the implementation being tested. 
I think it should test with the accessor connecting to the cassandra node in integration test. Please take `org.apache.cassandra.sidecar.db.RestoreSliceDatabaseAccessorIntTest` as an example.

Reply:
Added a different test